### PR TITLE
Add engine interaction model

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Adds a visuial description (BPMN Model) of the engine and streamprocessor interaction to the docs, in order to answer:  `How does the StreamPlatform and Engine interaction look like?`

Model:

engine-bpmn

 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: docs/engine_questions.md [hunk 1]
```diff
@@ -12,6 +12,13 @@ entity moving through the engine. Tokens are also not spawned and consumed for e
 * Activated elements
 * Taken sequence flows
 
+## How does the StreamPlatform and Engine interaction look like?
+
+Open the model and start the token flow to see the interaction between StreamProcessor (maintained by ZDP) and the RecordProcessor (Engine, maintained by the ZPA).
+
+[![engine-bpmn](https://user-images.githubusercontent.com/2758593/195591824-233ca579-96bc-4ee1-b6a3-5449929d75a9.png)](https://modeler.cloud.ultrawombat.com/diagrams/32c22fc0-709a-48d7-afe5-83a0c40a83fd--engine-bpmn?v=1076,651,1)
```
Comment (korthout) [inline docs/engine_questions.md:19]: I've created a milestone for the workflow at this moment, and made the change I suggested during the talk in a next milestone.

🔧 Please update the image to the latest state.

🔧 If you choose to ignore the above suggestion, please let me know so I follow up on it. IMO it makes sense to take it along now while we're changing this anyways.

**Review by korthout — APPROVED** [review]
Thanks @Zelldon 👏 

👍 I really like the model

🔧 I dislike a pull request without a description. Please add one before merging

🔧 Please consider my other suggestion, but if you don't please let me know

**Comment by Zelldon** [comment]
Ya well I created it quickly to not forgot about it, sorry. I have to take care of the kids, so idk I might update it tomorrow.

**Comment by korthout** [comment]
All good @Zelldon If not, I'll take it over on Monday 👍

File: docs/engine_questions.md (hunk 1 above)
Comment (Zelldon) [reply docs/engine_questions.md]: Thanks for that @korthout :+1: I also added my missing part regarding the acceptance call to the model. I updated the uploaded picture as well.

**Comment by oleschoenburg** [comment]
bors retry

**Pull request merged**